src/cmds/build_cmds.rs

```diff
@@ -0,0 +1,96 @@
+//! Commands for building zrc toolchains
+
+use clap::Parser;
+use std::error::Error;
+
+use crate::cli::DispatchCommand;
+use crate::{build, deps, git_utils, installer, paths};
+
+/// Build a specific version of zrc
+#[derive(Parser)]
+pub struct BuildCmd {
+    /// The git reference to build (branch, tag, or commit)
+    pub reference: String,
+
+    /// Custom zrc repository URL
+    #[arg(
+        long = "zrc-repo",
+        default_value = "https://github.com/zirco-lang/zrc.git"
+    )]
+    pub repo_url: String,
+}
+
+impl DispatchCommand for BuildCmd {
+    fn dispatch(self) -> Result<(), Box<dyn Error>> {
+        // Check dependencies first
+        deps::warn_dependencies();
+
+        // Ensure directories exist
+        paths::ensure_directories()?;
+
+        let source_dir = paths::zrc_source_dir();
+
+        // Clone or open repository
+        let repo = git_utils::clone_or_open(&self.repo_url, &source_dir)?;
+
+        // Fetch latest changes
+        git_utils::fetch(&repo)?;
+
+        // Checkout the requested reference
+        git_utils::checkout_ref(&repo, &self.reference)?;
+
+        // Get commit SHA for version naming
+        let commit_sha = git_utils::get_current_commit_short(&repo)?;
+
+        // Determine version name (use tag if it's a tag, otherwise use ref + commit)
+        let version = if self.reference.starts_with('v') && !self.reference.contains('/') {
```

Comment:
Is there a better way to check if it's a tag, commit, or a head?

I want

Tag: v0.1.0
Commit: <full SHA>
Head: main@<short SHA>

Currently:
Tag: v0.1.0
Commit: <full SHA>@<short SHA>
Head: Ok, but what happens if a commit SHA starts with a v, or worse, the tag is just 0.1.0

Reply:
Remove the explicit prefix.